californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java

```diff
@@ -365,6 +385,15 @@ public UdpMatcher(final NetworkConfig config) {
 			}
 			// ignore response
 			return null;
+		} else if (isResponseRelatedToMulticastRequest(exchange)) {
```

Comment:
I don't think that we need this separate block which mostly duplicates the code from the `else if (isResponseRelatedToRequest) {` block.
Instead, we could simply add another case to that other block checking for a multicast request:
````
} else if (isResponseRelatedToRequest(exchange, responseContext)) {
  Exchange prev = deduplicator.findPrevious(idByMID, exchange);
  Exchange prev = deduplicator.findPrevious(idByMID, exchange);
  if (prev != null) {
    LOGGER.log(Level.FINER, "Received duplicate response for open exchange: {0}", response);
    response.setDuplicate(true);
  } else if (exchange.getCurrentRequest().isMulticast()) {
    // keep exchange open because we might receive additional responses
    LOGGER.log(Level.FINER, "Received response for multicast request: {0}", response);
  } else {
    idByMID = new KeyMID(exchange.getCurrentRequest().getMID());
    exchangesByMID.remove(idByMID);
    LOGGER.log(Level.FINE, "Closed open request [{0}]", idByMID);
  }
  if (response.getType() == Type.ACK && exchange.getCurrentRequest().getMID() != response.getMID()) {
    LOGGER.log(Level.WARNING,
        "Possible MID reuse before lifetime end for token [{0}], expected MID {1} but received {2}",
        new Object[] { response.getTokenString(), exchange.getCurrentRequest().getMID(),
        response.getMID() });
  }
  return exchange;
}
````

WDYT?